core/xml.js

```diff
@@ -422,6 +422,9 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
             'another location.');
         }
         variablesFirst = false;
+      } else if (name == 'parsererror') {
```

Comment:
Wow....  [`<parsererror>`](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser#Error_handling)  I did not know about that.

That said... `name` comes from `xmlChild`, and wouldn't `<parsererror>` be the root element of the error document?

Further, this function doesn't do any parsing.  The DOM tree is passed in as a parameter, so its reasonable to assume it is a valid DOM. Move the test against `<parsererror>` into `textToDom(..)`.

Reply:
At least in Chrome, `<parsererror>` ends up being the first child node of the root node. The browser then renders as much of the document as possible.

Good point about moving the check to `textToDom`. I will update accordingly.